fastapi_events/otel/utils.py

```diff
@@ -39,6 +40,9 @@ def create_span_for_handle_fn(
 
     links, context = [], None
 
+    if isinstance(payload, BaseModel):
```

Comment:
Hi, thanks for your contribution :) 

Even though the package is called `fastapi_events`, I know there are some users using it without FastAPI and Pydantic. So, importing `pydantic` will break those use cases. 

Instead of importing BaseModel from pydantic. I suggest using duck typing to maximise backward compatibility. 

Also, IIRC, the `model_dump` method was introduced in Pydantic v2. It replaced the `dict()` method in v1.

I'd suggest something like:

```python
if hasattr(payload, "model_dump"):  # handle Pydantic v2
    payload = payload.model_dump()
elif hasattr(payload, "dict"):    # handles Pydantic v1
    payload = payload.dict()     
```

In fact, we have similar logic handling both pydantic versions and the absence of pydantic in the dispatcher:
- https://github.com/melvinkcx/fastapi-events/blob/f44705b4031435d42f6e34bcf635c5d69f8bfdff/fastapi_events/dispatcher.py#L22
- https://github.com/melvinkcx/fastapi-events/blob/f44705b4031435d42f6e34bcf635c5d69f8bfdff/fastapi_events/dispatcher.py#L156

Once you update the changes, I'll be sure to have it merged asap. Thanks again for your contribution :)

Reply:
I updated the changes as you recommended. Duck typing seemed less verbose in this case than duplicating the `HAS_PYDANTIC`... logic from the dispatcher, but i'd be fine with either approach.

Reply:
Thanks, it works for me